.github/workflows/image_build_push.yml

```diff
@@ -46,3 +38,40 @@ jobs:
     - name: push docker image
       run: |
         docker push $DOCKER_USER/${GITHUB_REPOSITORY#*/}:${GITHUB_REF##*/}_${{ steps.date.outputs.date }}
+
+    - name: Create a text file
+      run: |
+        echo ${{ steps.date.outputs.date }} > tag_file.txt
+        echo "Created tag text file"
+
+    - name: Upload Artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: docker-image-tag
+        path: tag_file.txt
+        overwrite: true
```

Comment:
why do we need the `tag_file.txt` to be uploaded here given that we are passing it directly to the workflows on line 77?

Reply:
See Mukul's [comment in the issue](https://github.com/e-mission/e-mission-docs/issues/1048#issuecomment-2090417572):

> **Why I chose to add artifact method as well?**
> 
> The issue I was facing was with fetching the latest timestamp for the image tag in case of a push event trigger. This is because in the workflow dispatch, the server workflow itself would trigger the workflows and hence was in a way connected to these workflows. However, push events would only trigger the specific workflow in that specific dashboard repository to build and push the image and hence would not be able to retrieve the image tag directly.
> 
> So, I utilized the artifact upload and download method to:
> 
> * upload the image timestamp as an artifact in the workflow run for future use.
> * download the uploaded artifact from the latest previously successful and completed workflow run in e-mission-server repo for a specific branch (currently set to `tags-combo-approach` but to be changed to `master` once changes are final).

Reply:
I do see now why we do not need the artifact for the push event. [Detailed comments](https://github.com/e-mission/em-public-dashboard/pull/125#discussion_r1716150918) made here in public-dash Redesign PR.

Reply:
@MukuFlash03 I utilize the artifacts to get the tags to the internal repo, so please don't remove them yet!!

Reply:
@nataliejschultz For now, I've tested removing only the `Download artifacts` step in the public-dashboard workflow. 
I don't think it is related to the `Upload artifacts` step that you added for internal repo. 
Neither am I making changes to the server workflow. 

Can you please confirm if in this case it is alright to remove the artifact dealing with Push event from the dashboard workflows?

Reply:
@MukuFlash03 sorry, I misunderstood what you were changing. It is okay with me to remove the `download artifacts` step!